### PR TITLE
Add AzureQueue storage

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add AzureQueue storage.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [x]  Allow: Integration Tests
- [x]  Allow: All Required Checks
- [x]  Allow: All Builds
- [x]  Woolen Wolfdog

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 619438eb0e5950935b2902ad7d0a4b1362a45395 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65458/619438eb0e5950935b2902ad7d0a4b1362a45395/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Cloud repo sync (only for Clickhouse employees)</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>❌ failure</td></tr>
<tbody>
</table>

File: src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.h [hunk 1]
```diff
@@ -5,25 +5,24 @@
...
 #include <Storages/StorageFactory.h>
 
 
 namespace DB
 {
-class S3QueueMetadata;
+class ObjectStorageQueueMetadata;
 
-class StorageS3Queue : public IStorage, WithContext
+class StorageObjectStorageQueue : public IStorage, WithContext
```
Comment (kssenii) [inline src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.h:19]: P.S. 90% of the diff is just a rename `S3Queue` -> `ObjectStorageQueue`

**Comment by nikitamikhaylov** [comment]
@kssenii We basically need to rewrite all the documentation now...

**Comment by kssenii** [comment]
@nikitamikhaylov why all? Previous settings with prefix "s3queue_" are still supported for compatibility. So need only to mention that without prefix it will also work + a section for AzureQueue.

File: src/Storages/ObjectStorageQueue/registerS3Queue.cpp
Comment (antonio2368) [inline src/Storages/ObjectStorageQueue/registerS3Queue.cpp:1]: rename the file

File: src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp [hunk 2]
```diff
@@ -71,35 +60,35 @@ namespace
...
 
-        if (!s3queue_settings.s3queue_processing_threads_num)
+        if (!queue_settings.processing_threads_num)
         {
-            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Setting `s3queue_processing_threads_num` cannot be set to zero");
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Setting `processing_threads_num` cannot be set to zero");
         }
 
-        if (!s3queue_settings.s3queue_enable_logging_to_s3queue_log.changed)
+        if (!queue_settings.enable_logging_to_s3queue_log.changed)
```
Comment (antonio2368) [inline src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp:76]: what about `enable_logging_to_azure_queue_log`

File: src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp [hunk 3]
```diff
@@ -343,25 +332,28 @@ std::shared_ptr<StorageS3QueueSource> StorageS3Queue::createSource(
     {
         object_storage->removeObject(StoredObject(path));
     };
-    auto s3_queue_log = s3queue_settings->s3queue_enable_logging_to_s3queue_log ? local_context->getS3QueueLog() : nullptr;
-    return std::make_shared<StorageS3QueueSource>(
+    auto system_queue_log = queue_settings->enable_logging_to_s3queue_log
```
Comment (antonio2368) [inline src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp:335]: so you can enable s3queue log for AzureQueue?

File: tests/integration/test_storage_s3_queue/test.py [hunk 4]
```diff
@@ -263,15 +302,31 @@ def test_delete_after_processing(started_cluster, mode):
...
+            AZURE_CONTAINER_NAME
+        )
+        objects_iterator = client.list_blobs(files_path)
+        for objects in objects_iterator:
+            assert False
 
 
-@pytest.mark.parametrize("mode", AVAILABLE_MODES)
-def test_failed_retry(started_cluster, mode):
+@pytest.mark.parametrize(
```
Comment (antonio2368) [inline tests/integration/test_storage_s3_queue/test.py:318]: you can write `@pytest.mark.parametrize` multiple times
each combination will be run
```py
@pytest.mark.parametrize(argnames: "mode", argvalues: ["unordered", "ordered"])
@pytest.mark.parametrize(argnames: "engine_name", argvalues: ["S3Queue", "AzureQueue"])
```

Comment (kssenii) [reply tests/integration/test_storage_s3_queue/test.py:318]: It does not seem to work for me:
```
E     File "/home/kssenii/workspace/ClickHouse1/tests/integration/test_storage_s3_queue/test.py", line 252
E       @pytest.mark.parametrize(argnames: "mode", argvalues: ["unordered", "ordered"])
E                                        ^
E   SyntaxError: invalid syntax
```

Comment (antonio2368) [reply tests/integration/test_storage_s3_queue/test.py:318]: I'm an idiot and copied the things my editor shows
```py
@pytest.mark.parametrize("mode", ["unordered", "ordered"])
@pytest.mark.parametrize("engine_name", ["S3Queue", "AzureQueue"])
```

**Review by antonio2368 — APPROVED** [review]

**Comment by kssenii** [comment]
> Integration tests (aarch64) [6/6] — fail: 1

`test_ssl_cert_authentication/test.py::test_native_wrong_cert`

-- not related to changes

**Pull request merged**